arangod/Graph/GraphManager.cpp

```diff
@@ -311,22 +311,16 @@ bool GraphManager::graphExists(std::string const& graphName) const {
 
   Result res = trx.begin();
 
-  if (!res.ok()) {
-    return false;
+  if (res.fail()) {
+    THROW_ARANGO_EXCEPTION(res);
   }
 
   OperationOptions options;
-  try {
-    OperationResult checkDoc = trx.document(StaticStrings::GraphCollection,
-                                            checkDocument.slice(), options);
-    if (checkDoc.fail()) {
-      trx.finish(checkDoc.result);
-      return false;
-    }
-    trx.finish(checkDoc.result);
-  } catch (...) {
-  }
-  return true;
+  OperationResult checkDoc = trx.document(StaticStrings::GraphCollection,
+                                          checkDocument.slice(), options);
+  std::ignore = trx.finish(checkDoc.result);
+
+  return checkDoc.ok();
```

Comment:
I think this can be simplified to just
```suggestion
  return trx.finish(checkDoc.result).ok();
```
This will slightly change the existing behavior, as now a failing "finish" call can also lead to a result of "false" being returned. This would only happen in case of error, but is probably even justified.